sonaranalyzer-dotnet/tests/SonarAnalyzer.UnitTest/SyntaxTrackers/ObjectCreationTrackerTest.cs

```diff
@@ -0,0 +1,98 @@
+﻿/*
+ * SonarAnalyzer for .NET
+ * Copyright (C) 2015-2019 SonarSource SA
+ * mailto: contact AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+using System.Linq;
+using FluentAssertions;
+using Microsoft.CodeAnalysis;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+using SonarAnalyzer.Common;
+using SonarAnalyzer.Helpers;
+using SonarAnalyzer.UnitTest.TestFramework;
+using CSharpSyntax = Microsoft.CodeAnalysis.CSharp.Syntax;
+using VBSyntax = Microsoft.CodeAnalysis.VisualBasic.Syntax;
+
+
+namespace SonarAnalyzer.UnitTest.Helpers
+{
+    [TestClass]
+    public class ObjectCreationTrackerTest
+    {
+        [TestMethod]
+        public void ConstArgumentForParameter_CS()
+        {
+            var testInput = @"
+namespace NS
+{
+  public class Base
+  {
+    private Base(string a, string b, bool c, int d, object e) {}
+
+    private void Usage(string notAConst)
+    {
+      new Base(notAConst, ""myConst"", true, 4, new object());
+    }
+  }
+}
+";
+            var context = CreateContext<CSharpSyntax.ObjectCreationExpressionSyntax>(testInput, AnalyzerLanguage.CSharp);
+            var tracker = new CSharpObjectCreationTracker(null, null);
+
+            tracker.ConstArgumentForParameter(context, "a").Should().BeNull();
+            tracker.ConstArgumentForParameter(context, "b").Should().Be("myConst");
+            tracker.ConstArgumentForParameter(context, "c").Should().Be(true);
+            tracker.ConstArgumentForParameter(context, "d").Should().Be(4);
+            tracker.ConstArgumentForParameter(context, "e").Should().BeNull();
+        }
+
+        [TestMethod]
+        public void ConstArgumentForParameter_VB()
+        {
+            var testInput = @"
+Namespace NS
+    Public Class Base
+        Sub New(ByVal a As String, ByVal b As String, ByVal c As Boolean, ByVal d As Integer, ByVal e As Object)
```

Comment:
ByVal is optional, I suggest to remove one of them for demonstration purposes.
Constant can be passed into ByRef as well, I suggest add one more parameter as ByRef with constant input.